### DISCUSSION TITLE
addBase with multiple @font-face declarations

### DISCUSSION THREAD

**Discussion by jaskra** [tailwindlabs/tailwindcss#11358] [Help] [answered]
I am trying to use local fonts vs linked Google Fonts (or Bunny or any external linked), and it seems I can add my @font-face to the plugins / addBase like : 

```
plugin(function ({ addBase }) {
  addBase({
    '@font-face': {
      fontFamily: 'FontOne',
      src: 'url("../fonts/FontOne-Regular.ttf") format("truetype")'
    }
    })
})
```
This works.

However, if I want to add a second font or additional font weight, I try this : 

```
plugin(function ({ addBase }) {
  addBase({
    '@font-face': {
      fontFamily: 'FontOne',
      src: 'url("../fonts/FontOne-Regular.ttf") format("truetype")'
    },
    '@font-face': {
      fontFamily: 'FontTwo',
      src: 'url("../fonts/FontTwo-Regular.ttf") format("truetype")'
    },
    })
})
```

And it doesn't seem to work for one of the fonts. When I look at the compiled CSS file, there is only one @font-face that got created. 

In this example it is _FontTwo_  as that was the last one listed.

I am thinking this is because having two of the same tag name / declaration is conflicting?

I am able to get it to work if I add a second "addBase" declaration like this : 

```
plugin(function ({ addBase }) {
  addBase({
    '@font-face': {
      fontFamily: 'FontOne',
      src: 'url("../fonts/FontOne-Regular.ttf") format("truetype")'
    },   
    }),
    addBase({ 
     '@font-face': {
        fontFamily: 'FontTwo',
        src: 'url("../fonts/FontTwo-Regular.ttf") format("truetype")'
      },
    })
})
```

Since this seems a bit hacky to me, I was curious if anyone else is has multiple @font-face declarations in AddBase plugins, and if so, is there a better way to do this?

Also, is there any known problem with having multiple addBase calls in a single _plugin(function({ addBase }))_ ?

**Comment by wongjn** [comment]
> I am thinking this is because having two of the same tag name / declaration is conflicting?

You are seeing this behavior because of how JavaScript objects work. You cannot have multiple values with the same key:

```js
{
  '@font-face': {
    fontFamily: 'FontOne',
    src: 'url("../fonts/FontOne-Regular.ttf") format("truetype")'
  },
  '@font-face': {
    fontFamily: 'FontTwo',
    src: 'url("../fonts/FontTwo-Regular.ttf") format("truetype")'
  },
}
```
is like

```js
{
  a: 'foo',
  a: 'bar',
}
```
Which would evaluate to
```js
{
  a: 'bar',
}
```

> Also, is there any known problem with having multiple `addBase` calls in a single `plugin(function({ addBase }))`?

Not that I know of. I believe having multiple `addBase()` calls in a single `plugin()` function is within specification and should work fine.

**Reply by jaskra** [reply]
Thanks for clarifying and as I suspected as well. After my post, I did discover that I can pass an array to @font-face and the build script is smart enough to chunk those out as individual @font-face blocks, so I think that would be the cleanest way to do it with one addBase, one @font-face, and an array of declarations inside.

**Comment by jaskra** [comment]
One additional thing, I just noticed I can pass an array to @font-face using only one addBase and that seems to work as well.

For example: 

```
plugin(function ({ addBase }) {
  addBase({
    '@font-face': [
    {
      fontFamily: 'FontOne',
      src: 'url("../fonts/FontOne-Regular.ttf") format("truetype")'
    },
    {
      fontFamily: 'FontTwo',
      src: 'url("../fonts/FontTwo-Regular.ttf") format("truetype")'
    }
    ]
    })
})
```

**Comment by adamwathan** [comment, marked as answer]
Any of these options are totally fine 👍 Personally I would do this in your CSS file and not bother with config/plugins at all:

```css
@font-face {
  font-family: 'FontOne';
  src: url("../fonts/FontOne-Regular.ttf") format("truetype");
}
@font-face {
  font-family: 'FontTwo';
  src: url("../fonts/FontTwo-Regular.ttf") format("truetype");
}

@tailwind base;
@tailwind components;
@tailwind utilities;
```

**Reply by jaskra** [reply]
Thanks Adam, that's a lot simpler!
